include/mbedtls/gcm.h

```diff
@@ -113,10 +113,9 @@ int mbedtls_gcm_setkey(mbedtls_gcm_context *ctx,
 /**
  * \brief           This function performs GCM encryption or decryption of a buffer.
  *
- * \note            For encryption, the output buffer can be the same as the
- *                  input buffer. For decryption, the output buffer cannot be
- *                  the same as input buffer. If the buffers overlap, the output
- *                  buffer must trail at least 8 Bytes behind the input buffer.
+ * \note            The output buffer \p output can be the same as the input
+ *                  buffer \p input. If \p output is greater than \p input, they
+ *                  cannot overlap.
```

Comment:
So this is preexisting, but we aren't testing anything regarding overlap for the GCM API. I'm uncomfortable making a stronger promise without adding tests to ensure that we won't break that stronger promise.